src/Settings/SettingsServiceProvider.php

```diff
@@ -20,12 +20,21 @@ class SettingsServiceProvider extends AbstractServiceProvider
      */
     public function register()
     {
+        $this->container->singleton(DefaultSettingsRepository::class);
+
+        $this->container->alias(DefaultSettingsRepository::class, 'flarum.settings.default');
+
         $this->container->singleton(SettingsRepositoryInterface::class, function (Container $container) {
-            return new MemoryCacheSettingsRepository(
-                new DatabaseSettingsRepository(
-                    $container->make(ConnectionInterface::class)
+            $defaults = $container->make(DefaultSettingsRepository::class);
+            $defaults->setInner(
```

Comment:
Maybe `inner` could be a constructor arg, for more consistency with `MemoryCacheSettingsRepository`?

Reply:
That's how I initially implemented it, but the thing that's different about `DefaultSettingsRepository` is that it's not only acting as a `SettingsRepositoryInterface` but also as a repository for default setting values. That means we need to define the class itself as a singleton so that we can extend it in the extender to add defaults, and a constructor arg inner wouldn't work with that.

Reply:
Maybe we could use some kind of mutable collection to store the underlying map? Although that's a lot of boilerplate, probably not worth it.

Reply:
Wait, that's a great idea, if we can't bind an array, we definitely could bind a collection, that would cleanup the default settings repository implementation.